src/include/access/fileam.h

```diff
@@ -63,6 +63,17 @@ typedef struct ExternalInsertDescData
 
 typedef ExternalInsertDescData *ExternalInsertDesc;
 
+/*
+ * ExternalSelectDescData is used for storing state related
+ * to selecting data from an external table.
+ */
+typedef struct ExternalSelectDescData
```

Comment:
That's an option but since we plan on passing other data like the type of query, etc to PXF I think it will be more maintainable to have this encapsulating struct so that we don't need to keep changing the method signatures.

What are your thoughts?